pkg/backup/item_backupper.go

```diff
@@ -454,30 +456,62 @@ func (ib *defaultItemBackupper) takePVSnapshot(obj runtime.Unstructured, log log
 		"ark.heptio.com/pv":     metadata.GetName(),
 	}
 
-	log.Info("Snapshotting PersistentVolume")
-	snapshotID, err := blockStore.CreateSnapshot(volumeID, pvFailureDomainZone, tags)
-	if err != nil {
-		// log+error on purpose - log goes to the per-backup log file, error goes to the backup
-		log.WithError(err).Error("error creating snapshot")
-		return errors.WithMessage(err, "error creating snapshot")
-	}
-
+	log.Info("Getting volume information")
 	volumeType, iops, err := blockStore.GetVolumeInfo(volumeID, pvFailureDomainZone)
 	if err != nil {
 		log.WithError(err).Error("error getting volume info")
 		return errors.WithMessage(err, "error getting volume info")
 	}
 
-	if ib.backupRequest.Status.VolumeBackups == nil {
-		ib.backupRequest.Status.VolumeBackups = make(map[string]*api.VolumeBackupInfo)
+	log.Info("Snapshotting PersistentVolume")
+	snapshot := newVolumeSnapshot(ib.backupRequest.Backup, volumeType, pvFailureDomainZone, location, iops)
+
+	snapshotID, createSnapErr := blockStore.CreateSnapshot(volumeID, pvFailureDomainZone, tags)
+	if createSnapErr != nil {
+		log.WithError(createSnapErr).Error("error creating snapshot")
+		snapshot.Status.Phase = api.VolumeSnapshotPhaseFailed
+	} else {
+		snapshot.Status.Phase = api.VolumeSnapshotPhaseCompleted
+		snapshot.Status.ProviderSnapshotID = snapshotID
 	}
 
-	ib.backupRequest.Status.VolumeBackups[name] = &api.VolumeBackupInfo{
-		SnapshotID:       snapshotID,
-		Type:             volumeType,
-		Iops:             iops,
-		AvailabilityZone: pvFailureDomainZone,
+	createdSnap, err := ib.volumeSnapshotClient.VolumeSnapshots(snapshot.Namespace).Create(snapshot)
+	if err == nil {
+		ib.backupRequest.VolumeSnapshots = append(ib.backupRequest.VolumeSnapshots, createdSnap)
 	}
 
-	return nil
+	// nil errors are automatically removed
+	return kubeerrs.NewAggregate([]error{
```

Comment:
I would do this a bit differently. On ln 479, `if err != nil`, then return this aggregate error. Return `nil` at the end.